// dsg-ng2-bs4-modal/bundles/ng2-bs4-modal.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
System.register("src/ng2-bs4-modal/components/modal-instance", ["rxjs/Observable", "rxjs/add/operator/map", "rxjs/add/observable/fromEvent"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    function booleanOrValue(value) {
        if (value === 'true')
            return true;
        else if (value === 'false')
            return false;
        return value;
    }
    function toPromise(observable) {
        return new Promise(function (resolve, reject) {
            observable.subscribe(function (next) {
                resolve(next);
            });
        });
    }
    var Observable_1, ModalInstance, ModalResult;
    return {
        setters: [
            function (Observable_1_1) {
                Observable_1 = Observable_1_1;
            },
            function (_1) {
            },
            function (_2) {
            }
        ],
        execute: function () {
            ModalInstance = (function () {
                function ModalInstance(element) {
                    this.element = element;
                    this.suffix = '.ng2-bs4-modal';
                    this.shownEventName = 'shown.bs.modal' + this.suffix;
                    this.hiddenEventName = 'hidden.bs.modal' + this.suffix;
                    this.visible = false;
                    this.init();
                }
                ModalInstance.prototype.open = function () {
                    return this.show();
                };
                ModalInstance.prototype.close = function () {
                    this.result = ModalResult.Close;
                    return this.hide();
                };
                ModalInstance.prototype.dismiss = function () {
                    this.result = ModalResult.Dismiss;
                    return this.hide();
                };
                ModalInstance.prototype.destroy = function () {
                    var _this = this;
                    return this.hide().then(function () {
                        if (_this.$modal) {
                            _this.$modal.data('bs.modal', null);
                            _this.$modal.remove();
                        }
                    });
                };
                ModalInstance.prototype.show = function () {
                    var promise = toPromise(this.shown);
                    this.resetData();
                    this.$modal.modal();
                    return promise;
                };
                ModalInstance.prototype.hide = function () {
                    if (this.$modal && this.visible) {
                        var promise = toPromise(this.hidden);
                        this.$modal.modal('hide');
                        return promise;
                    }
                    return Promise.resolve(this.result);
                };
                ModalInstance.prototype.init = function () {
                    var _this = this;
                    this.$modal = jQuery(this.element.nativeElement);
                    this.$modal.appendTo('body');
                    this.shown = Observable_1.Observable.fromEvent(this.$modal, this.shownEventName)
                        .map(function () {
                        _this.visible = true;
                    });
                    this.hidden = Observable_1.Observable.fromEvent(this.$modal, this.hiddenEventName)
                        .map(function () {
                        var result = (!_this.result || _this.result.valueOf() === ModalResult.None.valueOf())
                            ? ModalResult.Dismiss : _this.result;
                        _this.result = ModalResult.None;
                        _this.visible = false;
                        return result;
                    });
                };
                ModalInstance.prototype.resetData = function () {
                    this.$modal.removeData();
                    this.$modal.data('backdrop', booleanOrValue(this.$modal.attr('data-backdrop')));
                    this.$modal.data('keyboard', booleanOrValue(this.$modal.attr('data-keyboard')));
                };
                return ModalInstance;
            }());
            exports_1("ModalInstance", ModalInstance);
            (function (ModalResult) {
                ModalResult[ModalResult["None"] = 0] = "None";
                ModalResult[ModalResult["Close"] = 1] = "Close";
                ModalResult[ModalResult["Dismiss"] = 2] = "Dismiss";
            })(ModalResult || (ModalResult = {}));
            exports_1("ModalResult", ModalResult);
        }
    };
});
System.register("src/ng2-bs4-modal/components/modal", ["@angular/core", "src/ng2-bs4-modal/components/modal-instance"], function (exports_2, context_2) {
    "use strict";
    var __moduleName = context_2 && context_2.id;
    var core_1, modal_instance_1, ModalComponent, ModalSize;
    return {
        setters: [
            function (core_1_1) {
                core_1 = core_1_1;
            },
            function (modal_instance_1_1) {
                modal_instance_1 = modal_instance_1_1;
            }
        ],
        execute: function () {
            ModalComponent = (function () {
                function ModalComponent(element) {
                    var _this = this;
                    this.element = element;
                    this.overrideSize = null;
                    this.visible = false;
                    this.animation = true;
                    this.backdrop = true;
                    this.keyboard = true;
                    this.onClose = new core_1.EventEmitter(false);
                    this.onDismiss = new core_1.EventEmitter(false);
                    this.onOpen = new core_1.EventEmitter(false);
                    this.instance = new modal_instance_1.ModalInstance(this.element);
                    this.instance.hidden.subscribe(function (result) {
                        _this.visible = _this.instance.visible;
                        if (result === modal_instance_1.ModalResult.Dismiss)
                            _this.onDismiss.emit(undefined);
                    });
                    this.instance.shown.subscribe(function () {
                        _this.onOpen.emit(undefined);
                    });
                }
                Object.defineProperty(ModalComponent.prototype, "fadeClass", {
                    get: function () { return this.animation; },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ModalComponent.prototype, "dataKeyboardAttr", {
                    get: function () { return this.keyboard; },
                    enumerable: true,
                    configurable: true
                });
                Object.defineProperty(ModalComponent.prototype, "dataBackdropAttr", {
                    get: function () { return this.backdrop; },
                    enumerable: true,
                    configurable: true
                });
                ModalComponent.prototype.ngOnDestroy = function () {
                    return this.instance && this.instance.destroy();
                };
                ModalComponent.prototype.open = function (size) {
                    var _this = this;
                    var result = true;
                    if (this.onBeforeOpen) {
                        result = this.onBeforeOpen();
                    }
                    if (result) {
                        if (ModalSize.validSize(size))
                            this.overrideSize = size;
                        return this.instance.open().then(function () {
                            _this.visible = _this.instance.visible;
                        });
                    }
                    else {
                        return new Promise(function (resolve, reject) { return resolve(); });
                    }
                };
                ModalComponent.prototype.close = function () {
                    var _this = this;
                    var result = true;
                    if (this.onBeforeClose) {
                        result = this.onBeforeClose();
                    }
                    if (result) {
                        return this.instance.close().then(function () {
                            _this.onClose.emit(undefined);
                        });
                    }
                    else {
                        return new Promise(function (resolve, reject) { return resolve(); });
                    }
                };
                ModalComponent.prototype.dismiss = function (event) {
                    event.preventDefault();
                    event.stopPropagation();
                    var result = true;
                    if (this.onBeforeDismiss) {
                        result = this.onBeforeDismiss();
                    }
                    if (result) {
                        return this.instance.dismiss();
                    }
                    else {
                        return new Promise(function (resolve, reject) { return resolve(); });
                    }
                };
                ModalComponent.prototype.isSmall = function () {
                    return this.overrideSize !== ModalSize.Large
                        && this.size === ModalSize.Small
                        || this.overrideSize === ModalSize.Small;
                };
                ModalComponent.prototype.isLarge = function () {
                    return this.overrideSize !== ModalSize.Small
                        && this.size === ModalSize.Large
                        || this.overrideSize === ModalSize.Large;
                };
                return ModalComponent;
            }());
            __decorate([
                core_1.Input(),
                __metadata("design:type", Boolean)
            ], ModalComponent.prototype, "animation", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", Object)
            ], ModalComponent.prototype, "backdrop", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", Boolean)
            ], ModalComponent.prototype, "keyboard", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", String)
            ], ModalComponent.prototype, "size", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", Function)
            ], ModalComponent.prototype, "onBeforeClose", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", Function)
            ], ModalComponent.prototype, "onBeforeDismiss", void 0);
            __decorate([
                core_1.Input(),
                __metadata("design:type", Function)
            ], ModalComponent.prototype, "onBeforeOpen", void 0);
            __decorate([
                core_1.Output(),
                __metadata("design:type", core_1.EventEmitter)
            ], ModalComponent.prototype, "onClose", void 0);
            __decorate([
                core_1.Output(),
                __metadata("design:type", core_1.EventEmitter)
            ], ModalComponent.prototype, "onDismiss", void 0);
            __decorate([
                core_1.Output(),
                __metadata("design:type", core_1.EventEmitter)
            ], ModalComponent.prototype, "onOpen", void 0);
            __decorate([
                core_1.HostBinding('class.fade'),
                __metadata("design:type", Boolean),
                __metadata("design:paramtypes", [])
            ], ModalComponent.prototype, "fadeClass", null);
            __decorate([
                core_1.HostBinding('attr.data-keyboard'),
                __metadata("design:type", Boolean),
                __metadata("design:paramtypes", [])
            ], ModalComponent.prototype, "dataKeyboardAttr", null);
            __decorate([
                core_1.HostBinding('attr.data-backdrop'),
                __metadata("design:type", Object),
                __metadata("design:paramtypes", [])
            ], ModalComponent.prototype, "dataBackdropAttr", null);
            ModalComponent = __decorate([
                core_1.Component({
                    selector: 'modal',
                    host: {
                        'class': 'modal',
                        'role': 'dialog',
                        'tabindex': '-1'
                    },
                    template: "\n        <div class=\"modal-dialog\" [ngClass]=\"{ 'modal-sm': isSmall(), 'modal-lg': isLarge() }\">\n            <div class=\"modal-content\">\n                <ng-content></ng-content>\n            </div>\n        </div>\n    "
                }),
                __metadata("design:paramtypes", [core_1.ElementRef])
            ], ModalComponent);
            exports_2("ModalComponent", ModalComponent);
            ModalSize = (function () {
                function ModalSize() {
                }
                ModalSize.validSize = function (size) {
                    return size && (size === ModalSize.Small || size === ModalSize.Large);
                };
                return ModalSize;
            }());
            ModalSize.Small = 'sm';
            ModalSize.Large = 'lg';
            exports_2("ModalSize", ModalSize);
        }
    };
});
System.register("src/ng2-bs4-modal/components/modal-header", ["@angular/core", "src/ng2-bs4-modal/components/modal"], function (exports_3, context_3) {
    "use strict";
    var __moduleName = context_3 && context_3.id;
    var core_2, modal_1, ModalHeaderComponent;
    return {
        setters: [
            function (core_2_1) {
                core_2 = core_2_1;
            },
            function (modal_1_1) {
                modal_1 = modal_1_1;
            }
        ],
        execute: function () {
            ModalHeaderComponent = (function () {
                function ModalHeaderComponent(modal) {
                    this.modal = modal;
                    this.showClose = false;
                    this.headerClasses = {};
                }
                return ModalHeaderComponent;
            }());
            __decorate([
                core_2.Input('show-close'),
                __metadata("design:type", Boolean)
            ], ModalHeaderComponent.prototype, "showClose", void 0);
            __decorate([
                core_2.Input('header-classes'),
                __metadata("design:type", Object)
            ], ModalHeaderComponent.prototype, "headerClasses", void 0);
            ModalHeaderComponent = __decorate([
                core_2.Component({
                    selector: 'modal-header',
                    template: "\n        <div class=\"modal-header\" [ngClass]=\"headerClasses\">\n            <ng-content></ng-content>\n            <button *ngIf=\"showClose\" type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\" (click)=\"modal.dismiss($event)\">\n                <span aria-hidden=\"true\">&times;</span>\n            </button>\n        </div>\n    "
                }),
                __metadata("design:paramtypes", [modal_1.ModalComponent])
            ], ModalHeaderComponent);
            exports_3("ModalHeaderComponent", ModalHeaderComponent);
        }
    };
});
System.register("src/ng2-bs4-modal/components/modal-body", ["@angular/core", "src/ng2-bs4-modal/components/modal"], function (exports_4, context_4) {
    "use strict";
    var __moduleName = context_4 && context_4.id;
    var core_3, modal_2, ModalBodyComponent;
    return {
        setters: [
            function (core_3_1) {
                core_3 = core_3_1;
            },
            function (modal_2_1) {
                modal_2 = modal_2_1;
            }
        ],
        execute: function () {
            ModalBodyComponent = (function () {
                function ModalBodyComponent(modal) {
                    this.modal = modal;
                }
                return ModalBodyComponent;
            }());
            ModalBodyComponent = __decorate([
                core_3.Component({
                    selector: 'modal-body',
                    template: "\n        <div class=\"modal-body\">\n            <ng-content></ng-content>\n        </div>\n    "
                }),
                __metadata("design:paramtypes", [modal_2.ModalComponent])
            ], ModalBodyComponent);
            exports_4("ModalBodyComponent", ModalBodyComponent);
        }
    };
});
System.register("src/ng2-bs4-modal/components/modal-footer", ["@angular/core", "src/ng2-bs4-modal/components/modal"], function (exports_5, context_5) {
    "use strict";
    var __moduleName = context_5 && context_5.id;
    var core_4, modal_3, ModalFooterComponent;
    return {
        setters: [
            function (core_4_1) {
                core_4 = core_4_1;
            },
            function (modal_3_1) {
                modal_3 = modal_3_1;
            }
        ],
        execute: function () {
            ModalFooterComponent = (function () {
                function ModalFooterComponent(modal) {
                    this.modal = modal;
                    this.showDefaultButtons = false;
                }
                return ModalFooterComponent;
            }());
            __decorate([
                core_4.Input('show-default-buttons'),
                __metadata("design:type", Boolean)
            ], ModalFooterComponent.prototype, "showDefaultButtons", void 0);
            ModalFooterComponent = __decorate([
                core_4.Component({
                    selector: 'modal-footer',
                    template: "\n        <div class=\"modal-footer\">\n            <ng-content></ng-content>\n            <button *ngIf=\"showDefaultButtons\" type=\"button\" class=\"btn btn-default\" data-dismiss=\"modal\" (click)=\"modal.dismiss()\">Close</button>\n            <button *ngIf=\"showDefaultButtons\" type=\"button\" class=\"btn btn-primary\" (click)=\"modal.close()\">Save</button>\n        </div>\n    "
                }),
                __metadata("design:paramtypes", [modal_3.ModalComponent])
            ], ModalFooterComponent);
            exports_5("ModalFooterComponent", ModalFooterComponent);
        }
    };
});
System.register("src/ng2-bs4-modal/directives/autofocus", ["@angular/core", "src/ng2-bs4-modal/components/modal"], function (exports_6, context_6) {
    "use strict";
    var __moduleName = context_6 && context_6.id;
    var core_5, modal_4, AutofocusDirective;
    return {
        setters: [
            function (core_5_1) {
                core_5 = core_5_1;
            },
            function (modal_4_1) {
                modal_4 = modal_4_1;
            }
        ],
        execute: function () {
            AutofocusDirective = (function () {
                function AutofocusDirective(el, modal) {
                    var _this = this;
                    this.el = el;
                    this.modal = modal;
                    if (modal != null) {
                        this.modal.onOpen.subscribe(function () {
                            _this.el.nativeElement.focus();
                        });
                    }
                }
                return AutofocusDirective;
            }());
            AutofocusDirective = __decorate([
                core_5.Directive({
                    selector: '[autofocus]'
                }),
                __metadata("design:paramtypes", [core_5.ElementRef, modal_4.ModalComponent])
            ], AutofocusDirective);
            exports_6("AutofocusDirective", AutofocusDirective);
        }
    };
});
System.register("src/ng2-bs4-modal/module/ng2-bs4-modal.module", ["@angular/common", "@angular/core", "src/ng2-bs4-modal/directives/autofocus", "src/ng2-bs4-modal/components/modal-footer", "src/ng2-bs4-modal/components/modal-body", "src/ng2-bs4-modal/components/modal-header", "src/ng2-bs4-modal/components/modal"], function (exports_7, context_7) {
    "use strict";
    var __moduleName = context_7 && context_7.id;
    var common_1, core_6, autofocus_1, modal_footer_1, modal_body_1, modal_header_1, modal_5, ModalModule;
    return {
        setters: [
            function (common_1_1) {
                common_1 = common_1_1;
            },
            function (core_6_1) {
                core_6 = core_6_1;
            },
            function (autofocus_1_1) {
                autofocus_1 = autofocus_1_1;
            },
            function (modal_footer_1_1) {
                modal_footer_1 = modal_footer_1_1;
            },
            function (modal_body_1_1) {
                modal_body_1 = modal_body_1_1;
            },
            function (modal_header_1_1) {
                modal_header_1 = modal_header_1_1;
            },
            function (modal_5_1) {
                modal_5 = modal_5_1;
            }
        ],
        execute: function () {
            ModalModule = (function () {
                function ModalModule() {
                }
                return ModalModule;
            }());
            ModalModule = __decorate([
                core_6.NgModule({
                    imports: [
                        common_1.CommonModule
                    ],
                    declarations: [
                        modal_5.ModalComponent,
                        modal_header_1.ModalHeaderComponent,
                        modal_body_1.ModalBodyComponent,
                        modal_footer_1.ModalFooterComponent,
                        autofocus_1.AutofocusDirective
                    ],
                    exports: [
                        modal_5.ModalComponent,
                        modal_header_1.ModalHeaderComponent,
                        modal_body_1.ModalBodyComponent,
                        modal_footer_1.ModalFooterComponent,
                        autofocus_1.AutofocusDirective
                    ]
                })
            ], ModalModule);
            exports_7("ModalModule", ModalModule);
        }
    };
});
System.register("src/ng2-bs4-modal/ng2-bs4-modal", ["src/ng2-bs4-modal/components/modal", "src/ng2-bs4-modal/components/modal-header", "src/ng2-bs4-modal/components/modal-body", "src/ng2-bs4-modal/components/modal-footer", "src/ng2-bs4-modal/components/modal-instance", "src/ng2-bs4-modal/module/ng2-bs4-modal.module"], function (exports_8, context_8) {
    "use strict";
    var __moduleName = context_8 && context_8.id;
    var exportedNames_1 = {
        "ModalModule": true
    };
    function exportStar_1(m) {
        var exports = {};
        for (var n in m) {
            if (n !== "default" && !exportedNames_1.hasOwnProperty(n)) exports[n] = m[n];
        }
        exports_8(exports);
    }
    return {
        setters: [
            function (modal_6_1) {
                exportStar_1(modal_6_1);
            },
            function (modal_header_2_1) {
                exportStar_1(modal_header_2_1);
            },
            function (modal_body_2_1) {
                exportStar_1(modal_body_2_1);
            },
            function (modal_footer_2_1) {
                exportStar_1(modal_footer_2_1);
            },
            function (modal_instance_2_1) {
                exportStar_1(modal_instance_2_1);
            },
            function (ng2_bs4_modal_module_1_1) {
                exports_8({
                    "ModalModule": ng2_bs4_modal_module_1_1["ModalModule"]
                });
            }
        ],
        execute: function () {
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/components/modal-body.ngfactory", ["src/ng2-bs4-modal/components/modal-body", "@angular/core/src/linker/view", "@angular/core/src/linker/view_utils", "@angular/core/src/metadata/view", "@angular/core/src/linker/view_type", "@angular/core/src/change_detection/constants", "@angular/core/src/linker/component_factory", "src/ng2-bs4-modal/components/modal"], function (exports_9, context_9) {
    "use strict";
    var __moduleName = context_9 && context_9.id;
    var import0, import1, import3, import4, import5, import6, import7, import8, Wrapper_ModalBodyComponent, renderType_ModalBodyComponent_Host, View_ModalBodyComponent_Host0, ModalBodyComponentNgFactory, styles_ModalBodyComponent, renderType_ModalBodyComponent, View_ModalBodyComponent0;
    return {
        setters: [
            function (import0_1) {
                import0 = import0_1;
            },
            function (import1_1) {
                import1 = import1_1;
            },
            function (import3_1) {
                import3 = import3_1;
            },
            function (import4_1) {
                import4 = import4_1;
            },
            function (import5_1) {
                import5 = import5_1;
            },
            function (import6_1) {
                import6 = import6_1;
            },
            function (import7_1) {
                import7 = import7_1;
            },
            function (import8_1) {
                import8 = import8_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_ModalBodyComponent = (function () {
                function Wrapper_ModalBodyComponent(p0) {
                    this._changed = false;
                    this.context = new import0.ModalBodyComponent(p0);
                }
                Wrapper_ModalBodyComponent.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_ModalBodyComponent.prototype.ngOnDestroy = function () {
                };
                Wrapper_ModalBodyComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_ModalBodyComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_ModalBodyComponent.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_ModalBodyComponent.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_ModalBodyComponent;
            }());
            exports_9("Wrapper_ModalBodyComponent", Wrapper_ModalBodyComponent);
            renderType_ModalBodyComponent_Host = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, [], {});
            View_ModalBodyComponent_Host0 = (function (_super) {
                __extends(View_ModalBodyComponent_Host0, _super);
                function View_ModalBodyComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalBodyComponent_Host0, renderType_ModalBodyComponent_Host, import5.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalBodyComponent_Host0.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'modal-body', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
                    this.compView_0 = new View_ModalBodyComponent0(this.viewUtils, this, 0, this._el_0);
                    this._ModalBodyComponent_0_3 = new Wrapper_ModalBodyComponent(this.injectorGet(import8.ModalComponent, this.parentIndex));
                    this.compView_0.create(this._ModalBodyComponent_0_3.context);
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
                    return new import7.ComponentRef_(0, this, this._el_0, this._ModalBodyComponent_0_3.context);
                };
                View_ModalBodyComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import0.ModalBodyComponent) && (0 === requestNodeIndex))) {
                        return this._ModalBodyComponent_0_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalBodyComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
                    this._ModalBodyComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
                    this.compView_0.internalDetectChanges(throwOnChange);
                };
                View_ModalBodyComponent_Host0.prototype.destroyInternal = function () {
                    this.compView_0.destroy();
                };
                View_ModalBodyComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalBodyComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
                    if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
                };
                return View_ModalBodyComponent_Host0;
            }(import1.AppView));
            exports_9("ModalBodyComponentNgFactory", ModalBodyComponentNgFactory = new import7.ComponentFactory('modal-body', View_ModalBodyComponent_Host0, import0.ModalBodyComponent));
            styles_ModalBodyComponent = [];
            renderType_ModalBodyComponent = import3.createRenderComponentType('', 1, import4.ViewEncapsulation.None, styles_ModalBodyComponent, {});
            View_ModalBodyComponent0 = (function (_super) {
                __extends(View_ModalBodyComponent0, _super);
                function View_ModalBodyComponent0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalBodyComponent0, renderType_ModalBodyComponent, import5.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalBodyComponent0.prototype.createInternal = function (rootSelector) {
                    var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
                    this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
                    this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'modal-body'), null);
                    this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
                    this.projectNodes(this._el_1, 0);
                    this._text_3 = this.renderer.createText(this._el_1, '\n        ', null);
                    this._text_4 = this.renderer.createText(parentRenderNode, '\n    ', null);
                    this.init(null, (this.renderer.directRenderer ? null : [
                        this._text_0,
                        this._el_1,
                        this._text_2,
                        this._text_3,
                        this._text_4
                    ]), null);
                    return null;
                };
                return View_ModalBodyComponent0;
            }(import1.AppView));
            exports_9("View_ModalBodyComponent0", View_ModalBodyComponent0);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("node_modules/@angular/common/src/directives/ng_if.ngfactory", ["@angular/common/src/directives/ng_if", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_10, context_10) {
    "use strict";
    var __moduleName = context_10 && context_10.id;
    var import0, import1, import3, Wrapper_NgIf;
    return {
        setters: [
            function (import0_2) {
                import0 = import0_2;
            },
            function (import1_2) {
                import1 = import1_2;
            },
            function (import3_2) {
                import3 = import3_2;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgIf = (function () {
                function Wrapper_NgIf(p0, p1) {
                    this._changed = false;
                    this.context = new import0.NgIf(p0, p1);
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_NgIf.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgIf.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgIf.prototype.check_ngIf = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.ngIf = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgIf.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_NgIf.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgIf.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgIf.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgIf;
            }());
            exports_10("Wrapper_NgIf", Wrapper_NgIf);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/components/modal-footer.ngfactory", ["src/ng2-bs4-modal/components/modal-footer", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view", "@angular/core/src/linker/view_utils", "@angular/core/src/metadata/view", "@angular/core/src/linker/view_type", "@angular/core/src/change_detection/constants", "@angular/core/src/linker/component_factory", "src/ng2-bs4-modal/components/modal", "@angular/core/src/linker/view_container", "node_modules/@angular/common/src/directives/ng_if.ngfactory", "@angular/core/src/linker/template_ref", "@angular/common/src/directives/ng_if"], function (exports_11, context_11) {
    "use strict";
    var __moduleName = context_11 && context_11.id;
    var import0, import1, import2, import3, import5, import6, import7, import8, import9, import10, import11, import12, import13, Wrapper_ModalFooterComponent, renderType_ModalFooterComponent_Host, View_ModalFooterComponent_Host0, ModalFooterComponentNgFactory, styles_ModalFooterComponent, View_ModalFooterComponent1, View_ModalFooterComponent2, renderType_ModalFooterComponent, View_ModalFooterComponent0;
    return {
        setters: [
            function (import0_3) {
                import0 = import0_3;
            },
            function (import1_3) {
                import1 = import1_3;
            },
            function (import2_1) {
                import2 = import2_1;
            },
            function (import3_3) {
                import3 = import3_3;
            },
            function (import5_2) {
                import5 = import5_2;
            },
            function (import6_2) {
                import6 = import6_2;
            },
            function (import7_2) {
                import7 = import7_2;
            },
            function (import8_2) {
                import8 = import8_2;
            },
            function (import9_1) {
                import9 = import9_1;
            },
            function (import10_1) {
                import10 = import10_1;
            },
            function (import11_1) {
                import11 = import11_1;
            },
            function (import12_1) {
                import12 = import12_1;
            },
            function (import13_1) {
                import13 = import13_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_ModalFooterComponent = (function () {
                function Wrapper_ModalFooterComponent(p0) {
                    this._changed = false;
                    this.context = new import0.ModalFooterComponent(p0);
                    this._expr_0 = import1.UNINITIALIZED;
                }
                Wrapper_ModalFooterComponent.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_ModalFooterComponent.prototype.ngOnDestroy = function () {
                };
                Wrapper_ModalFooterComponent.prototype.check_showDefaultButtons = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.showDefaultButtons = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_ModalFooterComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_ModalFooterComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_ModalFooterComponent.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_ModalFooterComponent.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_ModalFooterComponent;
            }());
            exports_11("Wrapper_ModalFooterComponent", Wrapper_ModalFooterComponent);
            renderType_ModalFooterComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
            View_ModalFooterComponent_Host0 = (function (_super) {
                __extends(View_ModalFooterComponent_Host0, _super);
                function View_ModalFooterComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalFooterComponent_Host0, renderType_ModalFooterComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalFooterComponent_Host0.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'modal-footer', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
                    this.compView_0 = new View_ModalFooterComponent0(this.viewUtils, this, 0, this._el_0);
                    this._ModalFooterComponent_0_3 = new Wrapper_ModalFooterComponent(this.injectorGet(import9.ModalComponent, this.parentIndex));
                    this.compView_0.create(this._ModalFooterComponent_0_3.context);
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
                    return new import8.ComponentRef_(0, this, this._el_0, this._ModalFooterComponent_0_3.context);
                };
                View_ModalFooterComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import0.ModalFooterComponent) && (0 === requestNodeIndex))) {
                        return this._ModalFooterComponent_0_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalFooterComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
                    this._ModalFooterComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
                    this.compView_0.internalDetectChanges(throwOnChange);
                };
                View_ModalFooterComponent_Host0.prototype.destroyInternal = function () {
                    this.compView_0.destroy();
                };
                View_ModalFooterComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalFooterComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
                    if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
                };
                return View_ModalFooterComponent_Host0;
            }(import2.AppView));
            exports_11("ModalFooterComponentNgFactory", ModalFooterComponentNgFactory = new import8.ComponentFactory('modal-footer', View_ModalFooterComponent_Host0, import0.ModalFooterComponent));
            styles_ModalFooterComponent = [];
            View_ModalFooterComponent1 = (function (_super) {
                __extends(View_ModalFooterComponent1, _super);
                function View_ModalFooterComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
                    return _super.call(this, View_ModalFooterComponent1, renderType_ModalFooterComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer) || this;
                }
                View_ModalFooterComponent1.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.createRenderElement(this.renderer, null, 'button', new import3.InlineArray8(6, 'class', 'btn btn-default', 'data-dismiss', 'modal', 'type', 'button'), null);
                    this._text_1 = this.renderer.createText(this._el_0, 'Close', null);
                    var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_0));
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [
                        this._el_0,
                        this._text_1
                    ]), [disposable_0]);
                    return null;
                };
                View_ModalFooterComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalFooterComponent1.prototype.handleEvent_0 = function (eventName, $event) {
                    this.markPathToRootAsCheckOnce();
                    var result = true;
                    if ((eventName == 'click')) {
                        var pd_sub_0 = (this.parentView.context.modal.dismiss() !== false);
                        result = (pd_sub_0 && result);
                    }
                    return result;
                };
                return View_ModalFooterComponent1;
            }(import2.AppView));
            View_ModalFooterComponent2 = (function (_super) {
                __extends(View_ModalFooterComponent2, _super);
                function View_ModalFooterComponent2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
                    return _super.call(this, View_ModalFooterComponent2, renderType_ModalFooterComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer) || this;
                }
                View_ModalFooterComponent2.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.createRenderElement(this.renderer, null, 'button', new import3.InlineArray4(4, 'class', 'btn btn-primary', 'type', 'button'), null);
                    this._text_1 = this.renderer.createText(this._el_0, 'Save', null);
                    var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_0));
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [
                        this._el_0,
                        this._text_1
                    ]), [disposable_0]);
                    return null;
                };
                View_ModalFooterComponent2.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalFooterComponent2.prototype.handleEvent_0 = function (eventName, $event) {
                    this.markPathToRootAsCheckOnce();
                    var result = true;
                    if ((eventName == 'click')) {
                        var pd_sub_0 = (this.parentView.context.modal.close() !== false);
                        result = (pd_sub_0 && result);
                    }
                    return result;
                };
                return View_ModalFooterComponent2;
            }(import2.AppView));
            renderType_ModalFooterComponent = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_ModalFooterComponent, {});
            View_ModalFooterComponent0 = (function (_super) {
                __extends(View_ModalFooterComponent0, _super);
                function View_ModalFooterComponent0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalFooterComponent0, renderType_ModalFooterComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalFooterComponent0.prototype.createInternal = function (rootSelector) {
                    var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
                    this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
                    this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'modal-footer'), null);
                    this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
                    this.projectNodes(this._el_1, 0);
                    this._text_3 = this.renderer.createText(this._el_1, '\n            ', null);
                    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_1, null);
                    this._vc_4 = new import10.ViewContainer(4, 1, this, this._anchor_4);
                    this._TemplateRef_4_5 = new import12.TemplateRef_(this, 4, this._anchor_4);
                    this._NgIf_4_6 = new import11.Wrapper_NgIf(this._vc_4.vcRef, this._TemplateRef_4_5);
                    this._text_5 = this.renderer.createText(this._el_1, '\n            ', null);
                    this._anchor_6 = this.renderer.createTemplateAnchor(this._el_1, null);
                    this._vc_6 = new import10.ViewContainer(6, 1, this, this._anchor_6);
                    this._TemplateRef_6_5 = new import12.TemplateRef_(this, 6, this._anchor_6);
                    this._NgIf_6_6 = new import11.Wrapper_NgIf(this._vc_6.vcRef, this._TemplateRef_6_5);
                    this._text_7 = this.renderer.createText(this._el_1, '\n        ', null);
                    this._text_8 = this.renderer.createText(parentRenderNode, '\n    ', null);
                    this.init(null, (this.renderer.directRenderer ? null : [
                        this._text_0,
                        this._el_1,
                        this._text_2,
                        this._text_3,
                        this._anchor_4,
                        this._text_5,
                        this._anchor_6,
                        this._text_7,
                        this._text_8
                    ]), null);
                    return null;
                };
                View_ModalFooterComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import12.TemplateRef) && (4 === requestNodeIndex))) {
                        return this._TemplateRef_4_5;
                    }
                    if (((token === import13.NgIf) && (4 === requestNodeIndex))) {
                        return this._NgIf_4_6.context;
                    }
                    if (((token === import12.TemplateRef) && (6 === requestNodeIndex))) {
                        return this._TemplateRef_6_5;
                    }
                    if (((token === import13.NgIf) && (6 === requestNodeIndex))) {
                        return this._NgIf_6_6.context;
                    }
                    return notFoundResult;
                };
                View_ModalFooterComponent0.prototype.detectChangesInternal = function (throwOnChange) {
                    var currVal_4_0_0 = this.context.showDefaultButtons;
                    this._NgIf_4_6.check_ngIf(currVal_4_0_0, throwOnChange, false);
                    this._NgIf_4_6.ngDoCheck(this, this._anchor_4, throwOnChange);
                    var currVal_6_0_0 = this.context.showDefaultButtons;
                    this._NgIf_6_6.check_ngIf(currVal_6_0_0, throwOnChange, false);
                    this._NgIf_6_6.ngDoCheck(this, this._anchor_6, throwOnChange);
                    this._vc_4.detectChangesInNestedViews(throwOnChange);
                    this._vc_6.detectChangesInNestedViews(throwOnChange);
                };
                View_ModalFooterComponent0.prototype.destroyInternal = function () {
                    this._vc_4.destroyNestedViews();
                    this._vc_6.destroyNestedViews();
                };
                View_ModalFooterComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
                    if ((nodeIndex == 4)) {
                        return new View_ModalFooterComponent1(this.viewUtils, this, 4, this._anchor_4, this._vc_4);
                    }
                    if ((nodeIndex == 6)) {
                        return new View_ModalFooterComponent2(this.viewUtils, this, 6, this._anchor_6, this._vc_6);
                    }
                    return null;
                };
                return View_ModalFooterComponent0;
            }(import2.AppView));
            exports_11("View_ModalFooterComponent0", View_ModalFooterComponent0);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("node_modules/@angular/common/src/directives/ng_class.ngfactory", ["@angular/common/src/directives/ng_class", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view_utils"], function (exports_12, context_12) {
    "use strict";
    var __moduleName = context_12 && context_12.id;
    var import0, import1, import3, Wrapper_NgClass;
    return {
        setters: [
            function (import0_4) {
                import0 = import0_4;
            },
            function (import1_4) {
                import1 = import1_4;
            },
            function (import3_4) {
                import3 = import3_4;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_NgClass = (function () {
                function Wrapper_NgClass(p0, p1, p2, p3) {
                    this._changed = false;
                    this.context = new import0.NgClass(p0, p1, p2, p3);
                    this._expr_0 = import1.UNINITIALIZED;
                    this._expr_1 = import1.UNINITIALIZED;
                }
                Wrapper_NgClass.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_NgClass.prototype.ngOnDestroy = function () {
                };
                Wrapper_NgClass.prototype.check_klass = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.klass = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_NgClass.prototype.check_ngClass = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
                        this._changed = true;
                        this.context.ngClass = currValue;
                        this._expr_1 = currValue;
                    }
                };
                Wrapper_NgClass.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    if (!throwOnChange) {
                        this.context.ngDoCheck();
                    }
                    return changed;
                };
                Wrapper_NgClass.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_NgClass.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_NgClass.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_NgClass;
            }());
            exports_12("Wrapper_NgClass", Wrapper_NgClass);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/components/modal-header.ngfactory", ["src/ng2-bs4-modal/components/modal-header", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view", "@angular/core/src/linker/view_utils", "@angular/core/src/metadata/view", "@angular/core/src/linker/view_type", "@angular/core/src/change_detection/constants", "@angular/core/src/linker/component_factory", "src/ng2-bs4-modal/components/modal", "@angular/core/src/linker/view_container", "node_modules/@angular/common/src/directives/ng_class.ngfactory", "node_modules/@angular/common/src/directives/ng_if.ngfactory", "@angular/core/src/change_detection/differs/iterable_differs", "@angular/core/src/change_detection/differs/keyvalue_differs", "@angular/core/src/linker/element_ref", "@angular/core/src/linker/template_ref", "@angular/common/src/directives/ng_if", "@angular/common/src/directives/ng_class"], function (exports_13, context_13) {
    "use strict";
    var __moduleName = context_13 && context_13.id;
    var import0, import1, import2, import3, import5, import6, import7, import8, import9, import10, import11, import12, import13, import14, import15, import16, import17, import18, Wrapper_ModalHeaderComponent, renderType_ModalHeaderComponent_Host, View_ModalHeaderComponent_Host0, ModalHeaderComponentNgFactory, styles_ModalHeaderComponent, View_ModalHeaderComponent1, renderType_ModalHeaderComponent, View_ModalHeaderComponent0;
    return {
        setters: [
            function (import0_5) {
                import0 = import0_5;
            },
            function (import1_5) {
                import1 = import1_5;
            },
            function (import2_2) {
                import2 = import2_2;
            },
            function (import3_5) {
                import3 = import3_5;
            },
            function (import5_3) {
                import5 = import5_3;
            },
            function (import6_3) {
                import6 = import6_3;
            },
            function (import7_3) {
                import7 = import7_3;
            },
            function (import8_3) {
                import8 = import8_3;
            },
            function (import9_2) {
                import9 = import9_2;
            },
            function (import10_2) {
                import10 = import10_2;
            },
            function (import11_2) {
                import11 = import11_2;
            },
            function (import12_2) {
                import12 = import12_2;
            },
            function (import13_2) {
                import13 = import13_2;
            },
            function (import14_1) {
                import14 = import14_1;
            },
            function (import15_1) {
                import15 = import15_1;
            },
            function (import16_1) {
                import16 = import16_1;
            },
            function (import17_1) {
                import17 = import17_1;
            },
            function (import18_1) {
                import18 = import18_1;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_ModalHeaderComponent = (function () {
                function Wrapper_ModalHeaderComponent(p0) {
                    this._changed = false;
                    this.context = new import0.ModalHeaderComponent(p0);
                    this._expr_0 = import1.UNINITIALIZED;
                    this._expr_1 = import1.UNINITIALIZED;
                }
                Wrapper_ModalHeaderComponent.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_ModalHeaderComponent.prototype.ngOnDestroy = function () {
                };
                Wrapper_ModalHeaderComponent.prototype.check_showClose = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.showClose = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_ModalHeaderComponent.prototype.check_headerClasses = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
                        this._changed = true;
                        this.context.headerClasses = currValue;
                        this._expr_1 = currValue;
                    }
                };
                Wrapper_ModalHeaderComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_ModalHeaderComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_ModalHeaderComponent.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_ModalHeaderComponent.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_ModalHeaderComponent;
            }());
            exports_13("Wrapper_ModalHeaderComponent", Wrapper_ModalHeaderComponent);
            renderType_ModalHeaderComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
            View_ModalHeaderComponent_Host0 = (function (_super) {
                __extends(View_ModalHeaderComponent_Host0, _super);
                function View_ModalHeaderComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalHeaderComponent_Host0, renderType_ModalHeaderComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalHeaderComponent_Host0.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'modal-header', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
                    this.compView_0 = new View_ModalHeaderComponent0(this.viewUtils, this, 0, this._el_0);
                    this._ModalHeaderComponent_0_3 = new Wrapper_ModalHeaderComponent(this.injectorGet(import9.ModalComponent, this.parentIndex));
                    this.compView_0.create(this._ModalHeaderComponent_0_3.context);
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
                    return new import8.ComponentRef_(0, this, this._el_0, this._ModalHeaderComponent_0_3.context);
                };
                View_ModalHeaderComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import0.ModalHeaderComponent) && (0 === requestNodeIndex))) {
                        return this._ModalHeaderComponent_0_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalHeaderComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
                    this._ModalHeaderComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
                    this.compView_0.internalDetectChanges(throwOnChange);
                };
                View_ModalHeaderComponent_Host0.prototype.destroyInternal = function () {
                    this.compView_0.destroy();
                };
                View_ModalHeaderComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalHeaderComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
                    if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
                };
                return View_ModalHeaderComponent_Host0;
            }(import2.AppView));
            exports_13("ModalHeaderComponentNgFactory", ModalHeaderComponentNgFactory = new import8.ComponentFactory('modal-header', View_ModalHeaderComponent_Host0, import0.ModalHeaderComponent));
            styles_ModalHeaderComponent = [];
            View_ModalHeaderComponent1 = (function (_super) {
                __extends(View_ModalHeaderComponent1, _super);
                function View_ModalHeaderComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
                    return _super.call(this, View_ModalHeaderComponent1, renderType_ModalHeaderComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer) || this;
                }
                View_ModalHeaderComponent1.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.createRenderElement(this.renderer, null, 'button', new import3.InlineArray8(8, 'aria-label', 'Close', 'class', 'close', 'data-dismiss', 'modal', 'type', 'button'), null);
                    this._text_1 = this.renderer.createText(this._el_0, '\n                ', null);
                    this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'span', new import3.InlineArray2(2, 'aria-hidden', 'true'), null);
                    this._text_3 = this.renderer.createText(this._el_2, '', null);
                    this._text_4 = this.renderer.createText(this._el_0, '\n            ', null);
                    var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_0));
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [
                        this._el_0,
                        this._text_1,
                        this._el_2,
                        this._text_3,
                        this._text_4
                    ]), [disposable_0]);
                    return null;
                };
                View_ModalHeaderComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalHeaderComponent1.prototype.handleEvent_0 = function (eventName, $event) {
                    this.markPathToRootAsCheckOnce();
                    var result = true;
                    if ((eventName == 'click')) {
                        var pd_sub_0 = (this.parentView.context.modal.dismiss($event) !== false);
                        result = (pd_sub_0 && result);
                    }
                    return result;
                };
                return View_ModalHeaderComponent1;
            }(import2.AppView));
            renderType_ModalHeaderComponent = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_ModalHeaderComponent, {});
            View_ModalHeaderComponent0 = (function (_super) {
                __extends(View_ModalHeaderComponent0, _super);
                function View_ModalHeaderComponent0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalHeaderComponent0, renderType_ModalHeaderComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalHeaderComponent0.prototype.createInternal = function (rootSelector) {
                    var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
                    this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
                    this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'modal-header'), null);
                    this._NgClass_1_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers, this.parentIndex), this.parentView.injectorGet(import14.KeyValueDiffers, this.parentIndex), new import15.ElementRef(this._el_1), this.renderer);
                    this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
                    this.projectNodes(this._el_1, 0);
                    this._text_3 = this.renderer.createText(this._el_1, '\n            ', null);
                    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_1, null);
                    this._vc_4 = new import10.ViewContainer(4, 1, this, this._anchor_4);
                    this._TemplateRef_4_5 = new import16.TemplateRef_(this, 4, this._anchor_4);
                    this._NgIf_4_6 = new import12.Wrapper_NgIf(this._vc_4.vcRef, this._TemplateRef_4_5);
                    this._text_5 = this.renderer.createText(this._el_1, '\n        ', null);
                    this._text_6 = this.renderer.createText(parentRenderNode, '\n    ', null);
                    this.init(null, (this.renderer.directRenderer ? null : [
                        this._text_0,
                        this._el_1,
                        this._text_2,
                        this._text_3,
                        this._anchor_4,
                        this._text_5,
                        this._text_6
                    ]), null);
                    return null;
                };
                View_ModalHeaderComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import16.TemplateRef) && (4 === requestNodeIndex))) {
                        return this._TemplateRef_4_5;
                    }
                    if (((token === import17.NgIf) && (4 === requestNodeIndex))) {
                        return this._NgIf_4_6.context;
                    }
                    if (((token === import18.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 5)))) {
                        return this._NgClass_1_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalHeaderComponent0.prototype.detectChangesInternal = function (throwOnChange) {
                    var currVal_1_0_0 = 'modal-header';
                    this._NgClass_1_3.check_klass(currVal_1_0_0, throwOnChange, false);
                    var currVal_1_0_1 = this.context.headerClasses;
                    this._NgClass_1_3.check_ngClass(currVal_1_0_1, throwOnChange, false);
                    this._NgClass_1_3.ngDoCheck(this, this._el_1, throwOnChange);
                    var currVal_4_0_0 = this.context.showClose;
                    this._NgIf_4_6.check_ngIf(currVal_4_0_0, throwOnChange, false);
                    this._NgIf_4_6.ngDoCheck(this, this._anchor_4, throwOnChange);
                    this._vc_4.detectChangesInNestedViews(throwOnChange);
                };
                View_ModalHeaderComponent0.prototype.destroyInternal = function () {
                    this._vc_4.destroyNestedViews();
                };
                View_ModalHeaderComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
                    if ((nodeIndex == 4)) {
                        return new View_ModalHeaderComponent1(this.viewUtils, this, 4, this._anchor_4, this._vc_4);
                    }
                    return null;
                };
                return View_ModalHeaderComponent0;
            }(import2.AppView));
            exports_13("View_ModalHeaderComponent0", View_ModalHeaderComponent0);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/components/modal.ngfactory", ["src/ng2-bs4-modal/components/modal", "@angular/core/src/change_detection/change_detection_util", "@angular/core/src/linker/view", "@angular/core/src/linker/view_utils", "@angular/core/src/metadata/view", "@angular/core/src/linker/view_type", "@angular/core/src/change_detection/constants", "@angular/core/src/linker/component_factory", "@angular/core/src/linker/element_ref", "node_modules/@angular/common/src/directives/ng_class.ngfactory", "@angular/core/src/change_detection/differs/iterable_differs", "@angular/core/src/change_detection/differs/keyvalue_differs", "@angular/common/src/directives/ng_class"], function (exports_14, context_14) {
    "use strict";
    var __moduleName = context_14 && context_14.id;
    var import0, import1, import2, import3, import5, import6, import7, import8, import9, import10, import11, import12, import13, Wrapper_ModalComponent, renderType_ModalComponent_Host, View_ModalComponent_Host0, ModalComponentNgFactory, styles_ModalComponent, renderType_ModalComponent, View_ModalComponent0;
    return {
        setters: [
            function (import0_6) {
                import0 = import0_6;
            },
            function (import1_6) {
                import1 = import1_6;
            },
            function (import2_3) {
                import2 = import2_3;
            },
            function (import3_6) {
                import3 = import3_6;
            },
            function (import5_4) {
                import5 = import5_4;
            },
            function (import6_4) {
                import6 = import6_4;
            },
            function (import7_4) {
                import7 = import7_4;
            },
            function (import8_4) {
                import8 = import8_4;
            },
            function (import9_3) {
                import9 = import9_3;
            },
            function (import10_3) {
                import10 = import10_3;
            },
            function (import11_3) {
                import11 = import11_3;
            },
            function (import12_3) {
                import12 = import12_3;
            },
            function (import13_3) {
                import13 = import13_3;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_ModalComponent = (function () {
                function Wrapper_ModalComponent(p0) {
                    this._changed = false;
                    this.context = new import0.ModalComponent(p0);
                    this._expr_0 = import1.UNINITIALIZED;
                    this._expr_1 = import1.UNINITIALIZED;
                    this._expr_2 = import1.UNINITIALIZED;
                    this._expr_3 = import1.UNINITIALIZED;
                    this._expr_4 = import1.UNINITIALIZED;
                    this._expr_5 = import1.UNINITIALIZED;
                    this._expr_6 = import1.UNINITIALIZED;
                    this._expr_7 = import1.UNINITIALIZED;
                    this._expr_8 = import1.UNINITIALIZED;
                    this._expr_9 = import1.UNINITIALIZED;
                }
                Wrapper_ModalComponent.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_ModalComponent.prototype.ngOnDestroy = function () {
                    this.context.ngOnDestroy();
                    (this.subscription0 && this.subscription0.unsubscribe());
                    (this.subscription1 && this.subscription1.unsubscribe());
                    (this.subscription2 && this.subscription2.unsubscribe());
                };
                Wrapper_ModalComponent.prototype.check_animation = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
                        this._changed = true;
                        this.context.animation = currValue;
                        this._expr_0 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_backdrop = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
                        this._changed = true;
                        this.context.backdrop = currValue;
                        this._expr_1 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_keyboard = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
                        this._changed = true;
                        this.context.keyboard = currValue;
                        this._expr_2 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_size = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
                        this._changed = true;
                        this.context.size = currValue;
                        this._expr_3 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_onBeforeClose = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
                        this._changed = true;
                        this.context.onBeforeClose = currValue;
                        this._expr_4 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_onBeforeDismiss = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
                        this._changed = true;
                        this.context.onBeforeDismiss = currValue;
                        this._expr_5 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.check_onBeforeOpen = function (currValue, throwOnChange, forceUpdate) {
                    if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_6, currValue))) {
                        this._changed = true;
                        this.context.onBeforeOpen = currValue;
                        this._expr_6 = currValue;
                    }
                };
                Wrapper_ModalComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_ModalComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                    var currVal_7 = this.context.fadeClass;
                    if (import3.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
                        view.renderer.setElementClass(el, 'fade', currVal_7);
                        this._expr_7 = currVal_7;
                    }
                    var currVal_8 = this.context.dataKeyboardAttr;
                    if (import3.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
                        view.renderer.setElementAttribute(el, 'data-keyboard', ((currVal_8 == null) ? null : currVal_8.toString()));
                        this._expr_8 = currVal_8;
                    }
                    var currVal_9 = this.context.dataBackdropAttr;
                    if (import3.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
                        view.renderer.setElementAttribute(el, 'data-backdrop', ((currVal_9 == null) ? null : currVal_9.toString()));
                        this._expr_9 = currVal_9;
                    }
                };
                Wrapper_ModalComponent.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_ModalComponent.prototype.subscribe = function (view, _eventHandler, emit0, emit1, emit2) {
                    this._eventHandler = _eventHandler;
                    if (emit0) {
                        (this.subscription0 = this.context.onClose.subscribe(_eventHandler.bind(view, 'onClose')));
                    }
                    if (emit1) {
                        (this.subscription1 = this.context.onDismiss.subscribe(_eventHandler.bind(view, 'onDismiss')));
                    }
                    if (emit2) {
                        (this.subscription2 = this.context.onOpen.subscribe(_eventHandler.bind(view, 'onOpen')));
                    }
                };
                return Wrapper_ModalComponent;
            }());
            exports_14("Wrapper_ModalComponent", Wrapper_ModalComponent);
            renderType_ModalComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
            View_ModalComponent_Host0 = (function (_super) {
                __extends(View_ModalComponent_Host0, _super);
                function View_ModalComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
                    return _super.call(this, View_ModalComponent_Host0, renderType_ModalComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                }
                View_ModalComponent_Host0.prototype.createInternal = function (rootSelector) {
                    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'modal', new import3.InlineArray8(6, 'class', 'modal', 'role', 'dialog', 'tabindex', '-1'), rootSelector, null);
                    this.compView_0 = new View_ModalComponent0(this.viewUtils, this, 0, this._el_0);
                    this._ModalComponent_0_3 = new Wrapper_ModalComponent(new import9.ElementRef(this._el_0));
                    this.compView_0.create(this._ModalComponent_0_3.context);
                    this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
                    return new import8.ComponentRef_(0, this, this._el_0, this._ModalComponent_0_3.context);
                };
                View_ModalComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import0.ModalComponent) && (0 === requestNodeIndex))) {
                        return this._ModalComponent_0_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
                    this._ModalComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
                    this._ModalComponent_0_3.checkHost(this, this.compView_0, this._el_0, throwOnChange);
                    this.compView_0.internalDetectChanges(throwOnChange);
                };
                View_ModalComponent_Host0.prototype.destroyInternal = function () {
                    this.compView_0.destroy();
                    this._ModalComponent_0_3.ngOnDestroy();
                };
                View_ModalComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
                    cb(this._el_0, ctx);
                };
                View_ModalComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
                    if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
                };
                return View_ModalComponent_Host0;
            }(import2.AppView));
            exports_14("ModalComponentNgFactory", ModalComponentNgFactory = new import8.ComponentFactory('modal', View_ModalComponent_Host0, import0.ModalComponent));
            styles_ModalComponent = [];
            renderType_ModalComponent = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_ModalComponent, {});
            View_ModalComponent0 = (function (_super) {
                __extends(View_ModalComponent0, _super);
                function View_ModalComponent0(viewUtils, parentView, parentIndex, parentElement) {
                    var _this = _super.call(this, View_ModalComponent0, renderType_ModalComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways) || this;
                    _this._map_9 = import3.pureProxy2(function (p0, p1) {
                        return {
                            'modal-sm': p0,
                            'modal-lg': p1
                        };
                    });
                    return _this;
                }
                View_ModalComponent0.prototype.createInternal = function (rootSelector) {
                    var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
                    this._text_0 = this.renderer.createText(parentRenderNode, '\n        ', null);
                    this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'modal-dialog'), null);
                    this._NgClass_1_3 = new import10.Wrapper_NgClass(this.parentView.injectorGet(import11.IterableDiffers, this.parentIndex), this.parentView.injectorGet(import12.KeyValueDiffers, this.parentIndex), new import9.ElementRef(this._el_1), this.renderer);
                    this._text_2 = this.renderer.createText(this._el_1, '\n            ', null);
                    this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'div', new import3.InlineArray2(2, 'class', 'modal-content'), null);
                    this._text_4 = this.renderer.createText(this._el_3, '\n                ', null);
                    this.projectNodes(this._el_3, 0);
                    this._text_5 = this.renderer.createText(this._el_3, '\n            ', null);
                    this._text_6 = this.renderer.createText(this._el_1, '\n        ', null);
                    this._text_7 = this.renderer.createText(parentRenderNode, '\n    ', null);
                    this.init(null, (this.renderer.directRenderer ? null : [
                        this._text_0,
                        this._el_1,
                        this._text_2,
                        this._el_3,
                        this._text_4,
                        this._text_5,
                        this._text_6,
                        this._text_7
                    ]), null);
                    return null;
                };
                View_ModalComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
                    if (((token === import13.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 6)))) {
                        return this._NgClass_1_3.context;
                    }
                    return notFoundResult;
                };
                View_ModalComponent0.prototype.detectChangesInternal = function (throwOnChange) {
                    var currVal_1_0_0 = 'modal-dialog';
                    this._NgClass_1_3.check_klass(currVal_1_0_0, throwOnChange, false);
                    var currVal_1_0_1 = this._map_9(this.context.isSmall(), this.context.isLarge());
                    this._NgClass_1_3.check_ngClass(currVal_1_0_1, throwOnChange, false);
                    this._NgClass_1_3.ngDoCheck(this, this._el_1, throwOnChange);
                };
                return View_ModalComponent0;
            }(import2.AppView));
            exports_14("View_ModalComponent0", View_ModalComponent0);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/directives/autofocus.ngfactory", ["src/ng2-bs4-modal/directives/autofocus"], function (exports_15, context_15) {
    "use strict";
    var __moduleName = context_15 && context_15.id;
    var import0, Wrapper_AutofocusDirective;
    return {
        setters: [
            function (import0_7) {
                import0 = import0_7;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            Wrapper_AutofocusDirective = (function () {
                function Wrapper_AutofocusDirective(p0, p1) {
                    this._changed = false;
                    this.context = new import0.AutofocusDirective(p0, p1);
                }
                Wrapper_AutofocusDirective.prototype.ngOnDetach = function (view, componentView, el) {
                };
                Wrapper_AutofocusDirective.prototype.ngOnDestroy = function () {
                };
                Wrapper_AutofocusDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
                    var changed = this._changed;
                    this._changed = false;
                    return changed;
                };
                Wrapper_AutofocusDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
                };
                Wrapper_AutofocusDirective.prototype.handleEvent = function (eventName, $event) {
                    var result = true;
                    return result;
                };
                Wrapper_AutofocusDirective.prototype.subscribe = function (view, _eventHandler) {
                    this._eventHandler = _eventHandler;
                };
                return Wrapper_AutofocusDirective;
            }());
            exports_15("Wrapper_AutofocusDirective", Wrapper_AutofocusDirective);
        }
    };
});
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
System.register("src/ng2-bs4-modal/module/ng2-bs4-modal.module.ngfactory", ["@angular/core/src/linker/ng_module_factory", "src/ng2-bs4-modal/module/ng2-bs4-modal.module", "@angular/common/src/common_module", "@angular/common/src/localization", "@angular/core/src/i18n/tokens"], function (exports_16, context_16) {
    "use strict";
    var __moduleName = context_16 && context_16.id;
    var import0, import1, import2, import3, import5, ModalModuleInjector, ModalModuleNgFactory;
    return {
        setters: [
            function (import0_8) {
                import0 = import0_8;
            },
            function (import1_7) {
                import1 = import1_7;
            },
            function (import2_4) {
                import2 = import2_4;
            },
            function (import3_7) {
                import3 = import3_7;
            },
            function (import5_5) {
                import5 = import5_5;
            }
        ],
        execute: function () {/**
             * @fileoverview This file is generated by the Angular 2 template compiler.
             * Do not edit.
             * @suppress {suspiciousCode,uselessCode,missingProperties}
             */
            /* tslint:disable */
            ModalModuleInjector = (function (_super) {
                __extends(ModalModuleInjector, _super);
                function ModalModuleInjector(parent) {
                    return _super.call(this, parent, [], []) || this;
                }
                Object.defineProperty(ModalModuleInjector.prototype, "_NgLocalization_2", {
                    get: function () {
                        if ((this.__NgLocalization_2 == null)) {
                            (this.__NgLocalization_2 = new import3.NgLocaleLocalization(this.parent.get(import5.LOCALE_ID)));
                        }
                        return this.__NgLocalization_2;
                    },
                    enumerable: true,
                    configurable: true
                });
                ModalModuleInjector.prototype.createInternal = function () {
                    this._CommonModule_0 = new import2.CommonModule();
                    this._ModalModule_1 = new import1.ModalModule();
                    return this._ModalModule_1;
                };
                ModalModuleInjector.prototype.getInternal = function (token, notFoundResult) {
                    if ((token === import2.CommonModule)) {
                        return this._CommonModule_0;
                    }
                    if ((token === import1.ModalModule)) {
                        return this._ModalModule_1;
                    }
                    if ((token === import3.NgLocalization)) {
                        return this._NgLocalization_2;
                    }
                    return notFoundResult;
                };
                ModalModuleInjector.prototype.destroyInternal = function () {
                };
                return ModalModuleInjector;
            }(import0.NgModuleInjector));
            exports_16("ModalModuleNgFactory", ModalModuleNgFactory = new import0.NgModuleFactory(ModalModuleInjector, import1.ModalModule));
        }
    };
});
